behavior.c

```diff
@@ -397,7 +398,248 @@ static void pairing_tick(void) {
 		dbnc = 0;
 	}	
 	
-}	
+}
+
+static void setting_tick(void) {
+#define SET_MENU (-1)
+#define SET_VOLUME 0
+#define SET_MOTOR 1
+#define SET_RF_PARING 2
+#define SPEED_STEP 128
+#define BLINK 28
+	static unsigned char dbnc;
+	static unsigned char all_off;
+	static unsigned char setting_start;
+	static char setting_mode;
+	static char setting_select;
+	static char volume;
+	static char correction;
+	static unsigned char blink;
+	unsigned char ledrgb[3];
+
+	all_off = !(buttons_state[0] | buttons_state[1] | buttons_state[2]
+			| buttons_state[3] | buttons_state[4]);
+
+
+	if (setting_start == 1) {
+		// Exit case: Serial port open !
+		if (usb_uart_serial_port_open() || (rf_get_status() & RF_DATA_RX)) {
+			behavior_stop(B_SETTING);
+			leds_set_body_rgb(0, 0, 0);
+			leds_set_circle(0, 0, 0, 0, 0, 0, 0, 0);
+			leds_set_prox_h(0, 0, 0, 0, 0, 0, 0, 0);
+			vmVariables.target[0] = 0;
+			vmVariables.target[1] = 0;
+			if (rf_get_status() & RF_PAIRING_MODE)
+				rf_pairing_stop();
+			init_vm_mode();
+			return;
+		}
+
+		switch (setting_select) {
+			case SET_VOLUME:
+				leds_set_body_rgb(32, 15, 0);
+				break;
+			case SET_MOTOR:
+				leds_set_body_rgb(15, 32, 0);
+				break;
+			case SET_RF_PARING:
+				leds_set_body_rgb(15, 0, 32);
+				break;
+		}
+
+		if (setting_mode == SET_MENU) {
+			// blink rear prox leds
+			if (++blink == BLINK) {
+				leds_set_prox_h(32, 32, 32, 32, 32, 32, 32, 32);
+			}
+			if (blink > BLINK * 2) {
+				blink = 0;
+				leds_set_prox_h(0, 0, 0, 0, 0, 0, 0, 0);
+			}
+		} else {
+			int p = pulse_get();
+			p = pulse_get(); //pulse twice faster
+			p = pulse_get(); //pulse 3 time faster
+			leds_set_prox_h(p, p, p, p, p, p, p, p);
+			switch (setting_select) {
+				case SET_VOLUME: //(32, 15,0)
+					ledrgb[0] = (32 * p) >> 5;
+					ledrgb[1] = (15 * p) >> 5;
+					ledrgb[2] = 0;
+					break;
+				case SET_MOTOR://(15, 32,0)
+					ledrgb[0] = (15 * p) >> 5;
+					ledrgb[1] = (32 * p) >> 5;
+					ledrgb[2] = 0;
+					break;
+				case SET_RF_PARING://(15, 0,32)
+					ledrgb[0] = (15 * p) >> 5;
+					ledrgb[1] = 0;
+					ledrgb[2] = (32 * p) >> 5;
+					break;
+			}
+			leds_set_body_rgb(ledrgb[0], ledrgb[1], ledrgb[2]);
+		}
+
+		switch (setting_mode) {
+			case SET_MENU:
+				when(buttons_state[BUTTON_BACKWARD])
+			{
```

Comment:
Inconsistent placement of parenthesis. Please follow the convention used elsewhere (I guess there is one).